File: 프로그래머스/체육복.cpp
#include <string>
#include <vector>

using namespace std;
int have[100];

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    for(int i=0; i<=n; i++){
        have[i]=1;
    }
    for(int i=0; i<lost.size(); i++){
        have[lost[i]]--;
    }
    for(int i=0; i<reserve.size(); i++){
        have[reserve[i]]++;
    }
    for(int i=0; i<=n; i++){
        if(have[i]==0){
            if(have[i-1]==2){
                have[i]++;
                have[i+1]--;
            }
            else if(have[i+1]==2){
                have[i]++;
                have[i+1]--;
            }
        }
    }
    for(int i=1; i<=n; i++){
        if(have[i]!=0){
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/체육복.py
def solution(n, lost, reserve):
    answer = 0
    info = [1]*(n+1)
    for i in reserve:
        info[i]+=1
    for i in lost:
        info[i]-=1
    for i in range(1,n+1):
        if(i-1>0 and info[i-1]==0 and info[i]>=2):
            info[i-1]+=1
            info[i]-=1
        if(i+1<n+1 and info[i+1]==0 and info[i]>=2):
            info[i+1]+=1
            info[i]-=1
    for i in range(1,n+1):
        if(info[i]>=1):
            answer+=1
    return answer